plugins/module_utils/k8s/client.py

```diff
@@ -0,0 +1,251 @@
+        elif isinstance(kubeconfig, dict):
+            _requires_kubernetes_at_least("17.17.0")
+            kubernetes.config.load_kube_config_from_dict(
+                config_dict=kubeconfig, **optional_arg
+            )
+    else:
+        kubernetes.config.load_kube_config(config_file=None, **optional_arg)
+
+
+def _create_configuration(auth: Dict):
+    def auth_set(*names: list) -> bool:
+        return all(auth.get(name) for name in names)
+
+    if auth_set("host"):
+        # Removing trailing slashes if any from hostname
+        auth["host"] = auth.get("host").rstrip("/")
+
+    if auth_set("username", "password", "host") or auth_set("api_key", "host"):
+        # We have enough in the parameters to authenticate, no need to load incluster or kubeconfig
+        pass
+    elif auth_set("kubeconfig") or auth_set("context"):
+        try:
+            _load_config(auth)
+        except Exception as err:
+            raise err
+
+    else:
+        # First try to do incluster config, then kubeconfig
+        try:
+            kubernetes.config.load_incluster_config()
+        except kubernetes.config.ConfigException:
+            try:
+                _load_config(auth)
+            except Exception as err:
+                raise err
+
+    # Override any values in the default configuration with Ansible parameters
+    # As of kubernetes-client v12.0.0, get_default_copy() is required here
+    try:
+        configuration = kubernetes.client.Configuration().get_default_copy()
+    except AttributeError:
+        configuration = kubernetes.client.Configuration()
+
+    for key, value in iteritems(auth):
+        if key in AUTH_ARG_MAP.keys() and value is not None:
+            if key == "api_key":
+                setattr(
+                    configuration, key, {"authorization": "Bearer {0}".format(value)}
+                )
+            elif key == "proxy_headers":
+                headers = urllib3.util.make_headers(**value)
+                setattr(configuration, key, headers)
+            else:
+                setattr(configuration, key, value)
+
+    return configuration
+
+
+def _configuration_digest(configuration) -> str:
+    m = hashlib.sha256()
+    for k in AUTH_ARG_MAP:
+        if not hasattr(configuration, k):
+            v = None
+        else:
+            v = getattr(configuration, k)
+        if v and k in ["ssl_ca_cert", "cert_file", "key_file"]:
+            with open(str(v), "r") as fd:
+                content = fd.read()
+                m.update(content.encode())
+        else:
+            m.update(str(v).encode())
+    digest = m.hexdigest()
+
+    return digest
+
+
+def cache(func):
+    def wrapper(*args):
+        digest = _configuration_digest(*args)
+        if digest in _pool:
+            client = _pool[digest]
+            return client
+        else:
+            client = func(*args)
+            _pool[digest] = client
```

Comment:
```suggestion
            _pool[digest] = client
            return client
```

Reply:
Right, thanks!

Reply:
I added only one return statement outside if!

Reply:
That works, too.